core/src/main/java/com/novoda/merlin/service/ConnectivityChangesForwarder.java

```diff
@@ -0,0 +1,75 @@
+package com.novoda.merlin.service;
+
+import com.novoda.merlin.NetworkStatus;
+import com.novoda.merlin.receiver.ConnectivityChangeEvent;
+import com.novoda.merlin.registerable.bind.BindListener;
+import com.novoda.merlin.registerable.connection.ConnectListener;
+import com.novoda.merlin.registerable.disconnection.DisconnectListener;
+
+class ConnectivityChangesForwarder {
+
+    private final NetworkStatusRetriever networkStatusRetriever;
+    private final DisconnectListener disconnectListener;
+    private final ConnectListener connectListener;
+    private final BindListener bindListener;
+    private final HostPinger hostPinger;
```

Comment:
@rock3r is it worth renaming this? It's called a host pinger but it takes an `Endpoint`, might be better if it is consistent everywhere and it is called `EndpointPinger`.